### PR TITLE
NikitaK/Convert DMT5 Page  to TS 52237

### PR DESCRIPTION
**Pull request by NikitaK-deriv**
Changes:

-   DMT5 Page Convert to TS

## Type of change

-   [ ] Bug fix
-   [ ] New feature
-   [ ] Update feature
-   [x] Refactor code
-   [ ] Translation to code
-   [ ] Translation to crowdin
-   [ ] Script configuration
-   [ ] Improve performance
-   [ ] Style only
-   [ ] Dependency update
-   [ ] Documentation update
-   [ ] Release

### CODE REVIEW

File: src/pages/dmt5/_interest_free_trading.tsx [hunk 1]
```diff
@@ -129,7 +133,7 @@ const StyledImg = styled.img`
     }
 `
 
-const CheckedText = ({ children }) => (
+const CheckedText: React.FC<TextCheckerType> = ({ children }) => (
```
Comment (prince-deriv) [inline src/pages/dmt5/_interest_free_trading.tsx:136]: Just use `ReactElement` here and you wouldn't need to add the props.

File: src/pages/dmt5/_interest_free_trading.tsx [hunk 2]
```diff
@@ -179,4 +183,6 @@ const InterestFreeTrading = () => {
     )
 }
 
+InterestFreeTrading.propTypes = {}
+
```
Comment (prince-deriv) [inline src/pages/dmt5/_interest_free_trading.tsx:187]: No need for this. If the eslint is firing an error, add a this

```
/* eslint-disable react/prop-types */
```

at the top of the file, till we fix the proper ts config.

File: src/pages/dmt5/_flexibility.tsx [hunk 3]
```diff
@@ -26,7 +34,7 @@ const StyledSyntheticIcon = styled.img`
     ${BaseIconStyle}
 `
 
-const content = [
+const content: ContactType[] = [
```
Comment (prince-deriv) [inline src/pages/dmt5/_flexibility.tsx:37]: Isn't it Content not Contact?

File: src/pages/dmt5/_dmt5-video.tsx [hunk 4]
```diff
@@ -1,10 +1,15 @@
-import React from 'react'
+import React, { ReactElement } from 'react'
 import { graphql, useStaticQuery } from 'gatsby'
 import styled from 'styled-components'
 import { localize } from 'components/localization'
 import { Header, QueryImage } from 'components/elements'
 import device from 'themes/device.js'
 
+type StepType = {
```
Comment (prince-deriv) [inline src/pages/dmt5/_dmt5-video.tsx:8]: change the name to `StepProps`

File: src/pages/dmt5/_dbanner.tsx [hunk 5]
```diff
@@ -7,6 +7,14 @@ import { LinkButton } from 'components/form'
 import { localize } from 'components/localization'
 import device from 'themes/device.js'
 
+type PropsType = {
```
Comment (prince-deriv) [inline src/pages/dmt5/_dbanner.tsx:10]: Change to `DBannerProps`

File: src/pages/dmt5/_dbanner.tsx [hunk 6]
```diff
@@ -102,7 +110,13 @@ const StyledHeader = styled(Header)`
     }
 `
 
-const DBanner = ({ background_pattern, data, is_ppc, title, image_alt }) => {
+const DBanner = ({
+    background_pattern,
+    data,
+    is_ppc,
+    title,
+    image_alt,
```
Comment (prince-deriv) [inline src/pages/dmt5/_dbanner.tsx:118]: If the type declared to props is optional, please add a default value

```
    background_pattern = '',
    data ='',
    is_ppc=false,
    title='',
    image_alt='',

File: src/pages/dmt5/_interest_free_trading.tsx (hunk 2 above)
Comment (sean-binary) [reply src/pages/dmt5/_interest_free_trading.tsx:187]: Are you still experiencing this on your side too @prince-deriv ? Yashim and myself don't seem to get this error when we remove the propTypes and the import of the package on the file itself when converting.

Comment (prince-deriv) [reply src/pages/dmt5/_interest_free_trading.tsx]: @sean-binary this is still occurring to me in some cases. The strange thing is it's not consistent.

File: src/pages/dmt5/_flexibility.tsx [hunk 7]
```diff
@@ -125,7 +133,7 @@ const StyledText = styled(Text)`
 `
 
 const Flexibility = () => {
-    const { is_eu_country } = React.useContext(DerivStore)
+    const is_eu_country = React.useContext(DerivStore)
```
Comment (sean-binary) [inline src/pages/dmt5/_flexibility.tsx:136]: Any reason we are removing this?

File: src/pages/dmt5/_margin_calculator.tsx [hunk 8]
```diff
@@ -9,6 +9,18 @@ import { Localize, localize } from 'components/localization'
 import device from 'themes/device'
 import { DerivStore } from 'store'
 
+interface CalculatorProps {
+    children?: React.ReactNode
+    key?: number
+    index?: number
+    name: ReactElement
+    text: ReactElement
```
Comment (sean-binary) [inline src/pages/dmt5/_margin_calculator.tsx:17]: Please keep it consistent either by destructuring out `ReactElement` & `ReactNode` or calling `React.` on it.

File: src/pages/dmt5/_flexibility.tsx (hunk 7 above)
Comment (NikitaK-deriv) [reply src/pages/dmt5/_flexibility.tsx:136]: Any reason we are removing this?
> 
Screenshot 2021-12-29 at 15 52 20

Yes, think that ts warned me. Do I need to return {} ?

File: src/pages/dmt5/_margin_calculator.tsx (hunk 8 above)
Comment (NikitaK-deriv) [reply src/pages/dmt5/_margin_calculator.tsx:17]: Done @sean-binary. Decided to use React. to not to import methods every time

File: src/pages/dmt5/_flexibility.tsx [hunk 9]
```diff
@@ -10,6 +10,14 @@ import SyntheticIcon from 'images/svg/dmt5/synthetic.svg'
 import device from 'themes/device'
 import { DerivStore } from 'store'
 
+type ContentType = {
+    header?
+    text?
+    icon?
```
[the remark this answers is not in the document]
Comment (NikitaK-deriv) [reply src/pages/dmt5/_flexibility.tsx:16]: @sean-binary don't know how to explain it :) My bad

File: src/pages/dmt5/_flexibility.tsx (hunk 7 above)
Comment (NikitaK-deriv) [reply src/pages/dmt5/_flexibility.tsx]: Added {is_eu_country}

**Comment by sean-binary** [comment]
As flagged up by sonar cloud, please rename the component from `DtraderTabs` to `Dmt5Video` and `Tabs` to SideTabs`.

File: src/pages/dmt5/index.tsx [hunk 10]
```diff
@@ -23,7 +23,12 @@ import DMT5BG2 from 'images/svg/dmt5/dmt5-bg2.svg'
 import { size } from 'themes/device'
 import { isBrowser } from 'common/utility'
 
-const meta_attributes = {
+type MetaAttributesType = {
+    og_title: string
+    og_description: string
+}
+
+const meta_attributes: MetaAttributesType = {
```
Comment (sean-binary) [inline src/pages/dmt5/index.tsx:31]: This can be a common type to be used by all these instances. Could be in `src/types/page.types.ts` possibly.

Comment (NikitaK-deriv) [reply src/pages/dmt5/index.tsx]: Created folder and file the same as you wrote

**Review by sean-binary — APPROVED** [review]

**Pull request merged**